index.js

```diff
@@ -1,21 +1,29 @@
 'use strict';
 const PCancelable = require('p-cancelable');
 
-const selectorCache = new Map();
+const targetCache = new Map();
```

Comment:
This also works as a `WeakMap`

Reply:
You are right! That wasn't the case before though, the only map that existed initially is now the inner map of that `targetCache`, which still has to remain as a regular `Map` .

Reply:
No worries, I've just been trying to look for places where `WeakMap` can work :)